lib/vtls/mbedtls.c

```diff
@@ -358,6 +360,95 @@ set_ssl_version_min_max(struct Curl_cfilter *cf, struct Curl_easy *data)
   return result;
 }
 
+#ifdef MBEDTLS_TLS_ECJPAKE_WITH_AES_128_CCM_8
+static int
+mbed_cipher_suite_get_str(uint16_t id, char *buf, size_t buf_size,
+                          bool prefer_rfc)
+{
+  if(id == MBEDTLS_TLS_ECJPAKE_WITH_AES_128_CCM_8)
+    strcpy(buf, "TLS_ECJPAKE_WITH_AES_128_CCM_8");
```

Comment:
`strcpy` is not bound by `buf_size`, do something like `msnprintf(buf, buf_size, "%s", ...);` instead

Reply:
but then you'll get a truncated (broken) name if the buffer is too small.  It seems better to check if the name fits and return an error if it does not fit.

Reply:
Changed to `snprintf` and clarified in the comments of the wrapped function `Curl_cipher_suite_get_str` that  the caller is responsible to provide a large enough buffer to prevent string truncation. The max string length of a cipher suite name is predictable, so providing a buffer of 64 bytes should suffice.